### PR TITLE
Option to select xverse as default browser wallet in settings

### PR DESCRIPTION
**Pull request by victorkirov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background
When a user connects to an extension wallet, if there are multiple wallets implementing the sats connect interface, then the last wallet loaded would get all event messages. A user may want to force Xverse as the priority wallet to intercept these messages if the app using sats connect is using the default providers.

# 🔄 Changes
This adds a new setting toggle to set Xverse as the priority sats-connect wallet.

Note that the setting is stored in chrome storage so that it's available from the content-script directly without having to send messages around.

# 🖼 Screenshot / 📹 Video
image

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/hooks/useChromeLocalStorage.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react';
+
+export const useChromeLocalStorage = <T extends unknown>(key: string, defaultValue?: T) => {
```
[the remark this answers is not in the document]
Comment (victorkirov) [reply src/app/hooks/useChromeLocalStorage.ts:4]: Ah, that's awesome. Will use that instead.

File: src/app/utils/chromeStorage.ts [hunk 2]
```diff
@@ -19,35 +19,35 @@ class ChromeStorage {
...
           return reject(this.getError());
         }
-        return resolve(response[key]);
+        return resolve(response[key] ?? defaultValue);
       });
     });
   }
 
-  removeItem(key: string): Promise<any> {
+  removeItem(key: string): Promise<void> {
```
Comment (victorkirov) [inline src/app/utils/chromeStorage.ts:44]: I tried changing this and the setItem one to boolean, but the seedVault stuff started to complain. Checking the rest of the code, nothing was actually using the return value, so I removed it.

File: src/app/utils/chromeStorage.ts [hunk 3]
```diff
@@ -19,35 +19,35 @@ class ChromeStorage {
...
           return reject(this.getError());
         }
-        return resolve(true);
+        return resolve();
       });
     });
   }
 
-  getItem<T = any>(key: string): Promise<T> {
+  getItem<T = any, D = undefined>(key: string, defaultValue?: D): Promise<T | D> {
```
Comment (victorkirov) [inline src/app/utils/chromeStorage.ts:33]: By default, the storage interface returns an empty object if the key is not present, resulting in a response of undefined. The Seed Vault expects null though, so I had to add this default value.

**Review by m-aboelenein — APPROVED** [review]

**Comment by m-aboelenein** [comment]
> @m-aboelenein merged before it was tested!

I tested this one, and the change is pretty minimal we can retest with the RC

**Pull request merged**